tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleModuleProjectImporter.java

```diff
@@ -83,7 +83,11 @@ private boolean findGradleFile( File file )
     @Override
     public void importProject( String location, IProgressMonitor monitor ) throws CoreException
     {
-        GradleUtil.importGradleProject( new File( location ), monitor );
+        File projectLocation = new File( location );
+
+        GradleUtil.importGradleProject( projectLocation, monitor );
+
+        ModuleCoreUtil.addFacets( projectLocation, monitor );
```

Comment:
We only need to add facets if there are JSPs in the project.  No other projects will benefit from the facets.  Lets not add them to all projects which will unnecessarily slow things down.